hatchways/team-gingerbread
# FE/IN Availability Page

### What this PR does (required):
- addresses #84
- creates UI for AvailabilityTab in EditProfile
- for each day, displays the day and a form for choosing start and end time
- on changing the times (default start is 10 am, default end is 10 pm), the editAvailability helper makes a call to profile/edit to change the profile in the database (only availableTime attribute; added this attribute to profile model)

### Screenshots / Videos (front-end only):
![image](https://user-images.githubusercontent.com/78042694/144787067-711952fa-1848-436d-a74b-e482b9d1c98e.png)


### Any information needed to test this feature (required):
- run server and app
- create a new account, then go to MongoDB
- copy and paste the user id into the mockId variable in client/mocks/mockId.tsx
- login to your account
- navigate to the Availability Page (profile icon in navbar -> profile -> availability)
- the form should be populated with the default values (10 am/pm)
- change a few values, then reload the page
- the changes should be saved and displayed on this page (confirm via MongoDB)

### Any issues with the current functionality (optional):
- I had to base this PR on a pending PR (EditProfile integration) to get the EditProfile tab to appear without errors
- issues with importing Material UI icons
- the mock doesn't include this feature, but the user should be able to change availability in the future, not just the current week
- need to add validation for form (can't have start time later than end time)
- did not use Formik due to complications with passing value and onChange props to AvailabilityRow
